js/interactive-guides/microprofile-config/microprofile-config-callback.js

```diff
@@ -225,30 +225,57 @@ var microprofileConfigCallBack = (function() {
     };
 
     var __isParamInAnnotation = function(annotationParams) {
-        var allMatch = false; 
+        var allMatch = false;
         if (annotationParams.length === 2) {
             var param1 = annotationParams[0];
             var param2 = annotationParams[1];
-                  
+
             if ((param1 === "name=\"download_url\"" &&
                  param2 === "defaultValue=\"ftp:\/\/music.com\/us\/download\"") ||
                 (param2 === "name=\"download_url\"" &&
                  param1 === "defaultValue=\"ftp:\/\/music.com\/us\/download\"")) {
                 allMatch = true;
             }
         }
-        return allMatch;      
+        return allMatch;
     }
 
     var __checkInjectionEditorContent = function(content) {
         var annotationIsThere = false;
         var editorContentBreakdown = __getInjectionConfigContent(content);
         if (editorContentBreakdown !== null) {
             annotationIsThere = __isParamInAnnotation(editorContentBreakdown);
-        } 
+        }
         return annotationIsThere;
     };
 
+    var __saveServerXML = function() {
+      var stepName = stepContent.getCurrentStepName();
+      var content = contentManager.getEditorContents(stepName);
+      if (__checkMicroProfileConfigFeatureContent(content)) {
+          var stepName = stepContent.getCurrentStepName();
+          contentManager.markCurrentInstructionComplete(stepName);
+      } else {
+          // display error to fix it
+          __createErrorLinkForCallBack(stepName, true);
```

Comment:
Almost forget about this one. It should call editor.createErrorLinkForCallBack(true, __addMicroProfileConfigFeature);